SIS_import/rclone_setup.sh

```diff
@@ -1,38 +1,47 @@
 #!/bin/bash -x 
-
-if [ -z "${AWS_S3_BUCKET}" ]; then
-    AWS_S3_BUCKET="umich-tl-sis"
-    echo "aws_s3_bucket set to default value of umich-tl-sis"
+if [ -z "${PROCESS}" ]; then
+    PROCESS="sis_set_url"
+    echo "PROCESS value missing; set to default value of sis_set_url"
 fi
-echo "AWS_S3_BUCKET set to ${AWS_S3_BUCKET}"
 
-RCLONE_OPTS="--config /app/config/secrets/rclone.conf --cache-dir /tmp/.cache/rclone/"
+if [ "${PROCESS}" == "sis_set_url" ]; then
+    ## run SIS set url script
+    ruby /usr/src/app/sis_set_url.rb
 
-# rclone --config ./rclone.conf config show
-rclone ${RCLONE_OPTS} lsd ~/
+elif [ "${PROCESS}" == "sis_upload" ]; then
+    ## run SIS upload script
+    if [ -z "${AWS_S3_BUCKET}" ]; then
+        AWS_S3_BUCKET="umich-tl-sis"
+        echo "AWS_S3_BUCKET set to default value of umich-tl-sis"
+    fi
+    echo "AWS_S3_BUCKET set to ${AWS_S3_BUCKET}"
 
-rclone -vv ${RCLONE_OPTS} ls sftp:/OUTBOUND
+    RCLONE_OPTS="--config /app/config/secrets/rclone.conf --cache-dir /tmp/.cache/rclone/"
 
-# SIS upload
-## 1.copy SIS zip file from the SFTP server
-rclone -vv ${RCLONE_OPTS} --exclude '.*' copy sftp:/OUTBOUND/ /usr/src/app/data
+    # rclone --config ./rclone.conf config show
+    rclone ${RCLONE_OPTS} lsd ~/
 
+    rclone -vv ${RCLONE_OPTS} ls sftp:/OUTBOUND
 
-## 2. start with the sis_upload.rb
-ruby /usr/src/app/sis_upload.rb
+    # SIS upload
+    ## 1.copy SIS zip file from the SFTP server
+    rclone -vv ${RCLONE_OPTS} --exclude '.*' copy sftp:/OUTBOUND/ /usr/src/app/data
 
-if [ $? -eq 0 ] 
-then
-    ##3. archive the SIS zip file to AWS
-    rclone copy ${RCLONE_OPTS} --no-traverse --exclude '.*' /usr/src/app/data aws:${AWS_S3_BUCKET}/archive
 
-    ## 4. remove the SIS zip file from SFTP server
-    rclone -vv ${RCLONE_OPTS} delete sftp:/OUTBOUND/
+    ## 2. start with the sis_upload.rb
+    ruby /usr/src/app/sis_upload.rb
 
-else
-    echo "ERROR: There was a problem in SIS upload process. Please check the details in log file. "
-fi
+    if [ $? -eq 0 ] 
+    then
+        ##3. archive the SIS zip file to AWS
+        rclone copy ${RCLONE_OPTS} --no-traverse --exclude '.*' /usr/src/app/data aws:${AWS_S3_BUCKET}/archive
 
+        ## 4. remove the SIS zip file from SFTP server
+        rclone -vv ${RCLONE_OPTS} delete sftp:/OUTBOUND/
 
-## SIS set url script
-ruby /usr/src/app/sis_set_url.rb
+    else
+        echo "ERROR: There was a problem in SIS upload process. Please check the details in log file. "
+    fi
+else 
+   echo "Please set the environment variable PROCESS to either sis_set_url or sis_upload."
```

Comment:
This really isn't necessary as PROCESS will be set to a default, but this is fine.

Reply:
Default value is assigned only if the PROCESS value is missing.

PROCESS could be set to other values, though.

Reply:
Ah okay, makes sense.

I don't think we ever *will* do that but good check. Thanks!